### ISSUE TITLE
HttpLink to get external API

### ISSUE THREAD

**Issue by naman-kumar8** [graphql-dotnet/graphql-dotnet#2567]
### Environment (if relevant)

As I've used Apollo Angular to use `Url` and get external data of GraphQL now as I see everywhere that POCO is used to create model from scratch, but I want to use the `HttpLink` in .NET application to get data and use **query** and **mutation** like this for example.

```
let source$ = this.apollo.mutate({
      mutation: gql`mutation {
        update_users(where: {id: {_eq: "${id}"}}, _set: {name: "Chakkar"}) {
          returning {
            id
            name
            rocket
            timestamp
            twitter
          }
        }
      }`
    })
```
as the constructor in angular uses this method to get data in a component.
```
constructor(private apolloProvider: Apollo) {
    this.apollo = this.apolloProvider.use('xyz2');
  }
```
Module gets connection to external client.
```
providers: [{
    provide: APOLLO_NAMED_OPTIONS, // <-- Different from standard initialization
    useFactory(httpLink: HttpLink): NamedOptions {
      return {
          xyz: {
            cache: new InMemoryCache(),
            link: httpLink.create({
              uri: 'https://48p1r2roz4.sse.codesandbox.io',
            }),
          },
          xyz2: {
            cache: new InMemoryCache(),
            link: httpLink.create({            
              uri: 'https://api.spacex.land/graphql/',
            }),
          },
      };
    },
    deps: [HttpLink],
  }]
```

as I can see its possible using like this.
```
var schema = Schema.For(@"
      type Query {
        hello: String
      }
    ");
```

Now how do you use **mutations** and **query** in .NET GraphQL from external API rather than creating everything using models and code first.

**Comment by cyberhck** [comment]
I'm not sure if what you're asking is related to angular or that's just offtopic, but do you mean schema first? Not write model and use Schema.For to build out your whole schema?

Yes, that's possible it's called schema-first approach, you can find that in the first page of docs if I recall correctly :slightly_smiling_face:

**Comment by naman-kumar8** [comment]
@cyberhck not schema-first I don't want to create POCO models all together I want to use external API accessing the queries.
`uri: 'https://48p1r2roz4.sse.codesandbox.io'` suppose I want to access this URL query or make a mutation how do I do that in the dot net?

**Comment by cyberhck** [comment]
do you mean calling a graphql server to get your data? basically being a client and accessing the data? or you mean you want to build a graphql server without writing POCO?

**Comment by naman-kumar8** [comment]
yes that's what I want .. I guess this is the answer .. which I was looking for POCO required or not but I needed to access external API. 

https://github.com/graphql-dotnet/graphql-client

**Comment by cyberhck** [comment]
There are 3 ways to do this:
- The first way is to just use a http client and make a basic POST request, but this isn't recommended.
- Second way is to use untyped graphql clients which are flexible and you can write any query you want but you'll have to parse the items once you receive the response
- Third way is what I prefer, I try to generate client SDKs and that let's me work with strongly typed client without having to write POCO myself (this is what I recommend)

BTW in the body of this issue the place where you showed: Schema.For, it's for writing a graphql server not a client :)

I saw that you did find graphql-client which is the second option and is good as well, but if you need to work with returned data, you'll have to somehow parse it which most likely means writing poco for response.

**Comment by naman-kumar8** [comment]
Yes actually what I found mostly was with strongly typed. I know I have to do it later or I can use the anonymously typed responses like this. I prefer the shorter way using anonymous.

```
var graphQLResponse = await graphQLClient.SendQueryAsync(personAndFilmsRequest, () => new { person = new PersonType()} );
var personName = graphQLResponse.Data.person.Name;
```

But the main thing was that I needed was this `new GraphQLHttpClient("https://swapi.apis.guru/", new NewtonsoftJsonSerializer())` for client
in the whole picture what I was searching was server which you mentioned above.
